atom/browser/api/atom_api_desktop_capturer.cc

```diff
@@ -118,60 +169,22 @@ void OnRefreshFinishedTask(atom::api::DesktopCapturer* cap) {
         source.display_id = base::Int64ToString(device_id);
       }
     }
-  }
 #elif defined(OS_MACOSX)
-  // On Mac, the IDs across the APIs match.
-  for (auto& source : sources) {
-    if (source.media_list_source.id.type ==
-        content::DesktopMediaID::TYPE_SCREEN) {
+    // On Mac, the IDs across the APIs match.
+    for (auto& source : screen_sources) {
       source.display_id = base::Int64ToString(source.media_list_source.id.id);
     }
-  }
 #endif  // defined(OS_WIN)
-  // TODO(ajmacd): Add Linux support. The IDs across APIs differ but Chrome only
-  // supports capturing the entire desktop on Linux. Revisit this if individual
-  // screen support is added.
-
-  content::BrowserThread::PostTask(content::BrowserThread::UI, FROM_HERE,
-                                   base::Bind(EmitFinished, sources, cap));
-}
-
-}  // namespace
-
-namespace atom {
-
-namespace api {
-
-DesktopCapturer::DesktopCapturer(v8::Isolate* isolate) {
-  Init(isolate);
-  capture_thread_ = base::CreateSequencedTaskRunnerWithTraits(
-      {base::WithBaseSyncPrimitives(), base::MayBlock(),
-       base::TaskPriority::USER_VISIBLE});
-}
-
-DesktopCapturer::~DesktopCapturer() {}
-
-void DesktopCapturer::StartHandling(bool capture_window,
-                                    bool capture_screen,
-                                    const gfx::Size& thumbnail_size) {
-  capture_thread_->PostTask(
-      FROM_HERE, base::BindOnce(StartHandlingTask, capture_window,
-                                capture_screen, thumbnail_size, this));
-}
-
-void DesktopCapturer::OnSourceAdded(int index) {}
-
-void DesktopCapturer::OnSourceRemoved(int index) {}
-
-void DesktopCapturer::OnSourceMoved(int old_index, int new_index) {}
-
-void DesktopCapturer::OnSourceNameChanged(int index) {}
+    // TODO(ajmacd): Add Linux support. The IDs across APIs differ but Chrome
+    // only supports capturing the entire desktop on Linux. Revisit this if
+    // individual screen support is added.
+    std::move(screen_sources.begin(), screen_sources.end(),
+              std::back_inserter(captured_sources_));
+  }
 
-void DesktopCapturer::OnSourceThumbnailChanged(int index) {}
+  if (!capture_window_ && !capture_screen_)
```

Comment:
When a machine does not have any screen or window, no event would be emitted, this leads to the timeout of tests since the results is empty on CI machines.

Reply:
Removing this check still has a failing "responds to subsequent calls of different options" test in Linux CI, because `desktopCapturer.getSources({ types: ['window'] }, callback)` is not returning any result.

From the comments of other tests this test should probably just be disabled on Linux?

Reply:
> When a machine does not have any screen or window, no event would be emitted

Ah, missed that!

> Removing this check still has a failing "responds to subsequent calls of different options" test in Linux CI, 

Before this PR, the refresh method was called irrespective of whether sources were captured https://github.com/electron/electron/blob/master/chromium_src/chrome/browser/media/native_desktop_media_list.cc#L209, the upstream implementation only tries to schedule a new refresh when atleast one source is captured https://cs.chromium.org/chromium/src/chrome/browser/media/webrtc/native_desktop_media_list.cc?sq=package:chromium&g=0&l=312, hence this method would never be called. I am thinking of emitting a new observer event when the source list empty from `DesktopMediaListbase::UpdateSourceList` which would allow us handle this scenario, we can't move the refresh call outside the existing check because it would emit preemptive block thumbnail captures for aura windows.